### ISSUE TITLE
`ascii` codec can't encode character "\u2019" in position 7240

### ISSUE THREAD

**Issue by toobiwankenobi** [CellProfiler/CellProfiler#4375]
When I want to export my measurements to a spreadsheet with the module "ExportToSpreadsheet" I get the error
```{python}
Encountered unrecoverable error in ExportToSpreadsheet during post-processing: `ascii` codec can't encode character "\u2019" in position 7240: ordinal not in range(128)
```

 - **OS: macOS** (of importance, see comment below)
 - CellProfiler Version 4.1.3

The same error was [reported](https://forum.image.sc/t/errors-in-exporttospreasheet/19596) for an earlier CP version. I tried the recommended suggestions to solve the issue but it didn't help. 

**Data/Pipeline**
The example .tiffs and the acquisition metadata can be found [here](https://www.dropbox.com/sh/bbckvn52nsokh9b/AADfkSAKEUKEkOBeY9F05yIOa?dl=0). I used the [following](https://github.com/BodenmillerGroup/ImcSegmentationPipeline/blob/main/cp4_pipelines/3_measure_mask_basic.cppipe) CellProfiler Pipeline. This pipeline requires certain custom-made modules which can be found [here](https://github.com/BodenmillerGroup/ImcPluginsCP/tree/master/plugins). 

Any ideas?

**Comment by toobiwankenobi** [comment]
it worked using windows so it is a macOS related issue

**Comment by DavidStirling** [comment]
Thanks, we believe this is related to MacOS's tendency to use special quotation mark characters instead of `'`. Is it possible that one of your file paths or metadata fields might contain this?

**Comment by mohamadmilad** [comment]
> Thanks, we believe this is related to MacOS's tendency to use special quotation mark characters instead of `'`. Is it possible that one of your file paths or metadata fields might contain this?

I got the same error, also I checked all my paths, and none of the files in the paths contain `’`!

**Comment by DavidStirling** [comment]
@mohamadmilad Would you happen to have a pipeline/image set that reproduces this? It'd be handy to compare each instance.

**Comment by toobiwankenobi** [comment]
> Thanks, we believe this is related to MacOS's tendency to use special quotation mark characters instead of `'`. Is it possible that one of your file paths or metadata fields might contain this?

Hi @DavidStirling , sorry for my late response. It is indeed a macOS related problem, using a windows machine "solved" the issue.

**Comment by mohamadmilad** [comment]
> @mohamadmilad Would you happen to have a pipeline/image set that reproduces this? It'd be handy to compare each instance.

I used windows and did not get that error!

**Comment by bethac07** [comment]
If we can't reproduce on our end, another thing you might try would be to make a copy of the pipeline and try deleting the modules one or a few at a time (start from the bottom and delete a couple (other than ExportToSpreadsheet of course), see if you get the error, etc), to try to narrow down what module specifically is causing it, to see if either a) that module writes something weird or b) if that module has any unusual characters, in say, its module notes.  I would say there's a decent-but-not-certain chance that the module notes are the issue; once you find the offending module, if it has notes you could delete them and see if that gets rid of the issue.  If so, please do let us know which character it is, and we will be sure to try to add a handler.

**Comment by nilseling** [comment]
Hi all, I was able to reproduce the error on macOS using the pipelines provided by @toobiwankenobi. As @DavidStirling suggested, the issue comes from special quotation marks in the Description field of some of the modules. I was able to narrow it down to the step in the `ExportToSpreadsheet` module when the Experiment information is written out. After deleting all quotation marks  (https://github.com/BodenmillerGroup/ImcSegmentationPipeline/pull/70) the error disappeared.
